app/controllers/marking_schemes_controller.rb

```diff
@@ -12,7 +12,24 @@ def index
   end
 
   def populate
-    render json: get_table_json_data
+    table = JSON.parse(get_table_json_data)
+    assignment_weights = []
+    spreadsheet_weights = []
+    table.map do |scheme|
```

Comment:
This loop doesn't make sense. You can generating columns; the number of columns should depend only on the number of assignments and spreadsheets, and nothing else. If you can't do this easily from the given data, don't use it and instead write your own entirely new query.